subsys/bluetooth/audio/csip_crypto.c

```diff
@@ -101,12 +99,12 @@ int bt_csip_sih(const uint8_t sirk[BT_CSIP_SET_SIRK_SIZE], uint32_t r,
 	 * result of sih.
 	 */
 
-	LOG_DBG("BE: res %s", bt_hex(res, sizeof(res)));
+	LOG_DBG("res %s", bt_hex(res, 16));
 
 	/* Result is the lowest 3 bytes */
-	*out = sys_get_be24(res + 13);
+	memcpy(out, res + 13, BT_CSIP_SIH_HASH_SIZE);
```

Comment:
Since we are using `bt_encrypt_be`, don't we need to swap this to make it little endian? Otherwise it seems like we are taking little endian as input and then big endian as output?

Reply:
Yes, you are right